packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -64,11 +64,20 @@ class RenderParagraph extends RenderBox {
   TextSpan get text => _textPainter.text;
   set text(TextSpan value) {
     assert(value != null);
-    if (_textPainter.text == value)
-      return;
-    _textPainter.text = value;
-    _overflowShader = null;
-    markNeedsLayout();
+    switch (_textPainter.text.compareTo(value)) {
+      case RenderComparison.identical:
+      case RenderComparison.metadata:
+        return;
```

Comment:
accident of how i wrote the code, though i can justify it post-hoc: the "return" is an early return (nothing changed, bail out), whereas the other two branches are actually doing something. If we ever add something after the switch, I'd expect the identical/metadata cases to do nothing, but the other cases to do it.